### PR TITLE
adding cjs tests

### PR DESCRIPTION
**Pull request by luu-alex**
## Description
#5942
Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../blob/1.x/CONTRIBUTIONS.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3/test/black_box/test/fixtures/system_tests_utils.ts [hunk 1]
```diff
@@ -0,0 +1,473 @@
+/*
```
Comment (luu-alex) [inline packages/web3/test/black_box/test/fixtures/system_tests_utils.ts]: Was receiving a modules error for this file in blackbox testing after the CJS/ESM PR so we include it in the blackbox tests now

File: packages/web3/test/black_box/test/cjs_tests/batchRequest.test.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+import { getSystemTestProvider, closeOpenConnection, isWs } from '../fixtures/system_tests_utils';
```
Comment (jdevcs) [inline packages/web3/test/black_box/test/cjs_tests/batchRequest.test.ts]: cjs_tests and esm_tests should both be different by having their own package.json and :

`"type": "module"` for esm blackbox in package json , 
`"type": "commonjs"` for cjs blackbox package.json

also each should have its own jest-config and `ts-jest` extending to respective cjs and esm tsconfig files. Like https://github.com/web3/web3.js/blob/8da5d8331e18271d573f498fb046aa87fc096dd2/packages/web3/test/config/jest.config.js#L4 is extending ./test/tsconfig.json , that is extending ../tsconfig.cjs.json

File: packages/web3/test/black_box/test/fixtures/system_tests_utils.ts (hunk 1 above)
Comment (avkos) [reply packages/web3/test/black_box/test/fixtures/system_tests_utils.ts]: Is it possible to use link to the file here?

**Review by avkos — CHANGES_REQUESTED** [review]

File: packages/web3/test/black_box/test/fixtures/system_tests_utils.ts (hunk 1 above)
Comment (luu-alex) [reply packages/web3/test/black_box/test/fixtures/system_tests_utils.ts:1]: I don't believe so, the blackbox tests cannot find the module when I symlink. Talked with junaid and he said its okay to duplicate. Maybe i should create an issue about this?
This is the [error](https://github.com/web3/web3.js/actions/runs/4625519650/jobs/8181477873)
```
Cannot find module 'web3/test/shared_fixtures/system_tests_utils' from 'test/web3-eth-accounts/hashMessage.test.ts'
web3:       18 | // the test takes a long time to run afterAll
web3:       19 | import Web3 from 'web3';
web3:     > 20 | import {
web3:          | ^
web3:       21 | 	closeOpenConnection,
web3:       22 | 	getSystemTestProvider,
web3:       23 | 	isWs,
web3:       at Resolver._throwModNotFoundError (node_modules/jest-resolve/build/resolver.js:491:11)
web3:       at Object.<anonymous> (test/web3-eth-accounts/hashMessage.test.ts:20:1)
```

Comment (luu-alex) [reply packages/web3/test/black_box/test/fixtures/system_tests_utils.ts]: yes will fix

File: packages/web3/test/black_box/test/cjs_tests/batchRequest.test.ts (hunk 2 above)
Comment (luu-alex) [reply packages/web3/test/black_box/test/cjs_tests/batchRequest.test.ts]: updated

**Review by jdevcs — APPROVED** [review]
Can you trigger these tests in CI, and before merging make sure all these are passing @luu-alex 
Thanks

File: packages/web3/scripts/black_box_test.sh [hunk 3]
```diff
@@ -30,7 +30,27 @@ echo "RPC client Protocol: " $MODE
...
 export WEB3_SYSTEM_TEST_BACKEND=$BACKEND
 
-cd test/black_box
+cd test/esm_black_box
+yarn --update-checksums
+yarn
+
+if [[ ${BACKEND} == "infura" ]]
+then
+    if [ ! $INFURA_HTTP ] || [ ! $INFURA_WSS ]
```
Comment (nikoulai) [inline packages/web3/scripts/black_box_test.sh]:
```suggestion
    if [ ! $INFURA_HTTP ] && [ ! $INFURA_WSS ]
```
Maybe we want to exit if both are empty or miss something?

File: packages/web3/scripts/black_box_test.sh [hunk 4]
```diff
@@ -30,7 +30,27 @@ echo "RPC client Protocol: " $MODE
...
+    then
+        echo "No Infura provider URL specified"
+        exit 1
+    elif [ $MODE == "http" ]
+    then
+        WEB3_SYSTEM_TEST_PROVIDER=$INFURA_HTTP yarn "test:$BACKEND:$MODE"
```
Comment (nikoulai) [inline packages/web3/scripts/black_box_test.sh]:
```suggestion
       WEB3_SYSTEM_TEST_PROVIDER=$INFURA_HTTP
    else
       WEB3_SYSTEM_TEST_PROVIDER=$INFURA_WSS
    fi
fi 
yarn "test:$BACKEND:$MODE"

```

**Review by nikoulai — APPROVED** [review]

File: packages/web3/scripts/black_box_test.sh (hunk 3 above)
Comment (luu-alex) [reply packages/web3/scripts/black_box_test.sh]: switching operands will mean tests will never run

File: packages/web3/scripts/black_box_test.sh (hunk 4 above)
Comment (luu-alex) [reply packages/web3/scripts/black_box_test.sh]: done, thanks!

**Review by avkos — APPROVED** [review]

**Pull request merged**